uniconnect-epfl/uniconnect
# fix: refine events

# What I did

I did a few things related to the Event type and Event loading:
- Remove the Date type on the events. This is done in order for the event to be serializable and thus be able to be passed as props in some specific screens. Indeed, a React Date is not serializable.
- make the events load once added
- add a list of participants to the event type
- add the creator of the event in the event type

# How I did it
- I added a hook for refreshing the page on a swipe down from the user.
- When an event is loaded or fetched from the database, it is a string in the ISO format. When we want to display it, we create a Date object. I had to change the query selector too.

# How to verify it
For the swipe down, you can go to the Home page, create an event, and then swipe down as in the video. This should automatically display your event.

# Demo video

[

https://github.com/uniconnect-epfl/uniconnect/assets/91184289/caf00ccb-e763-4ff3-85ed-ac6ea69114c7

](url)
# Pre-merge checklist
The changes I have introduced:
